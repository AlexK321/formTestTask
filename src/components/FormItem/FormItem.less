form {
  background-color: white;
  border-radius: 4px;
  padding: 24px 40px!important;
  width: 100%;
  height: 400px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;

  .ant-col {
    width: 100%!important;
    max-width: 100%!important;
  }

  .form-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    p {
      width: 60%;
      margin: auto;
    }
    .ant-row {
      width: 48.5%!important;
      max-width: 100%!important;
    }
  }

  input {
    border-radius: 4px!important;
    padding: 25px 18px 4px 11px!important;
  }

  button {
    border: 1px solid #00b4e3!important;
    background-color: #00b4e3!important;
    border-radius: 8px!important;
    color: #fff!important;
    padding: 0 50px!important;
    height: 60px!important;
    
    &:hover {
      border-color: #17cfff!important;
      background-color: #17cfff!important;
    }
  }
}  