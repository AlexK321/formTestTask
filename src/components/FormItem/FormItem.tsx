import React, { FC } from 'react';
import { Form, Input } from 'antd';
import './FormItem.css';
import { Rule } from 'antd/lib/form';

interface Props {
  name?: string;
  placeholder?: string;
  rules?: Rule[]; // remove any
}

const DEFAULT_RULES = [
  { required: true, message: 'Поле не должно быть пустым' },
  {
    max: 10,
    message: `Максимальная длинна строки - 10 символов`,
  },
];

//webstorm - Ctr+alt+O - vscode analogs

const FormItem: FC<Props> = ({ name, placeholder, rules = DEFAULT_RULES }) => (
  <Form.Item name={name} rules={rules}>
    <Input name={name} placeholder={placeholder} />
  </Form.Item>
);

export default FormItem;
