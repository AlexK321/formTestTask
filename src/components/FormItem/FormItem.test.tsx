import React from 'react';
import { shallow } from 'enzyme';
import FormItem from './FormItem';

describe('Component: FormItem', () => {
  const component = shallow(<FormItem />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
