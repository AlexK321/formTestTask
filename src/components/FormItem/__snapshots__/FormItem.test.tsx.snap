// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormItem should match snapshot 1`] = `
<FormItem
  rules={
    Array [
      Object {
        "message": "Поле не должно быть пустым",
        "required": true,
      },
      Object {
        "max": 10,
        "message": "Максимальная длинна строки - 10 символов",
      },
    ]
  }
>
  <Input
    type="text"
  />
</FormItem>
`;
